data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PluginSetting.java

```diff
@@ -88,49 +88,83 @@ public Object getAttributeOrDefault(final String attribute, final Object default
      * @param attribute    name of the attribute
      * @param defaultValue default value for the setting
      * @return the value of the specified attribute, or {@code defaultValue} if this settings contains no value for
-     * the attribute
+     * the attribute. If the value is null, null will be returned.
      */
     public Integer getIntegerOrDefault(final String attribute, final int defaultValue) {
-        return ((Number) getAttributeOrDefault(attribute, defaultValue)).intValue();
+        Object object = getAttributeOrDefault(attribute, defaultValue);
+        if (object == null) {
+            return null;
+        } else if (object instanceof Number) {
+            return ((Number) object).intValue();
+        } else if (object instanceof String) {
+            return Integer.valueOf(String.valueOf(object));
+        }
+
+        throw new IllegalArgumentException("Unexpected type for attribute: " + attribute);
```

Comment:
Can the invalid class be added to the error message with `attribute.getClass()`?